registry/fileconfig.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package registry
+
+import (
+	"encoding/json"
+	"os"
+)
+
+// [Glossery]
+// registry: The service that serves container images.
+// domain: The DNS domain name of a registry.
+// repository: The domain + complete path of an image (no tag).
+// index: The domain[+[sub]path] of an image (no tag) (i.e. repository prefix).
+// image: A complete domain + path + tag
+// credential: A base64 encoded username + password associated with an index.
+
+// CredentialsFile is the struct that represents the contents of the "auth.json" file
+// that stores or references credentials that enable authentication into specific
+// registries or even repositories.
+//
+// reference: https://www.mankier.com/5/containers-auth.json
+// alternate: https://man.archlinux.org/man/containers-auth.json.5
+type CredentialsFile struct {
+	Auths map[string]EncAuth `json:"auths"`
+
+	// CredHelpers is currently not supported by the podman task driver
+	// CredHelpers map[string]string  `json:"credHelpers"`
```

Comment:
Just to make sure I understand things, these credential helpers are the same concepts as the [plugin config cred helper](https://github.com/hashicorp/nomad-driver-podman/pull/265/files?diff=split&w=0#diff-0e426a43248661127a0c0ee115aef7a1093b635f8993b3f7ebb1dd9f05b8f249R138) (a script that is executed to return a username and password), but are different because these are defined in the `auth.json` file, so in order to support them we will have to parse them from the JSON file and run something like [`authFromCredsHelper`](https://github.com/hashicorp/nomad-driver-podman/pull/265/files?diff=split&w=0#diff-bc2e0486c538e9f1c2b65ab5b564fbd9c2def7fd43bf85439d38a33ecf8d7ba4R44) on them?

Reply:
That's exactly correct!

The main advantage to implementing support for specifying helpers in the `auth.json` file is that you can directly correlate a registry/image to a specific helper.

However, you could also just implement your own credsHelper wrapper that does the same mapping, and invokes the correct helper given a specific registry.